errors/middleware.go

```diff
@@ -0,0 +1,55 @@
+package errors
+
+import (
+	"net/http"
+
+	"bytes"
+
+	"github.com/sirupsen/logrus"
+)
+
+// errorReponseWriter - wrapper to ResponseWriter
+type errorReponseWriter struct {
+	http.ResponseWriter
+	status int
+	body   *bytes.Buffer
+}
+
+func (erw *errorReponseWriter) Write(data []byte) (int, error) {
+	if erw.status == 0 || erw.status >= 500 {
+		return erw.body.Write(data)
+	}
+	return erw.ResponseWriter.Write(data)
+}
+
+func (erw *errorReponseWriter) WriteHeader(code int) {
+	erw.status = code
+	erw.ResponseWriter.WriteHeader(code)
+}
+
+// Middleware -
+type Middleware interface {
+	ServeHTTP(w http.ResponseWriter, r *http.Request, next http.HandlerFunc)
+}
+
+type middleware struct {
+	logger *logrus.Logger
+}
+
+// NewMiddleware -
+func NewMiddleware(l *logrus.Logger) Middleware {
+	return &middleware{l}
+}
+
+func (mw *middleware) ServeHTTP(w http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+	eWriter := &errorReponseWriter{w, http.StatusOK, bytes.NewBuffer([]byte{})}
+	next(eWriter, r)
+
+	entry := logrus.NewEntry(mw.logger)
+
+	if eWriter.status == http.StatusInternalServerError {
+		entry.Error(eWriter.body.String())
+		eWriter.ResponseWriter.Write([]byte(http.StatusText(http.StatusInternalServerError)))
```

Comment:
Looks like the response body is going to have one format for `4xx` errors and a whole different one (just `Internal Server Error`) for `500`. For the latter, the response should at least have `Content-Type: text/plain`, right?

Besides that, I believe that we should have a default structured JSON error format, containing a little bit more of information.